# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../log4cxx/gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix to build with gcc >= 4.3

diff -ruN apache-log4cxx-0.10.0-orig/src/examples/cpp/console.cpp apache-log4cxx-0.10.0/src/examples/cpp/console.cpp
--- apache-log4cxx-0.10.0-orig/src/examples/cpp/console.cpp	2008-04-01 00:34:52.000000000 +0200
+++ apache-log4cxx-0.10.0/src/examples/cpp/console.cpp	2010-08-16 13:35:44.908007262 +0200
@@ -15,7 +15,8 @@
  * limitations under the License.
  */
 
-#include <stdlib.h>
+#include <cstdlib>
+#include <cstring>
 #include <log4cxx/logger.h>
 #include <log4cxx/consoleappender.h>
 #include <log4cxx/simplelayout.h>
diff -ruN apache-log4cxx-0.10.0-orig/src/main/cpp/inputstreamreader.cpp apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp
--- apache-log4cxx-0.10.0-orig/src/main/cpp/inputstreamreader.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp	2010-08-16 13:35:44.908007262 +0200
@@ -21,6 +21,8 @@
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
diff -ruN apache-log4cxx-0.10.0-orig/src/main/cpp/socketoutputstream.cpp apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp
--- apache-log4cxx-0.10.0-orig/src/main/cpp/socketoutputstream.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2010-08-16 13:35:44.908007262 +0200
@@ -20,6 +20,8 @@
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
